### PR TITLE
draft

### PR DESCRIPTION
**Pull request by cansibi**

### CODE REVIEW

**Comment by Qiming-Liu** [comment]
1. send email utils function, send reset password email.
2. the link in the email should be like `https://<domain>/reset-password?token=<token>`
3. the token should have a `enableChangePassword` property
4. only when the api got `enableChangePassword = true` from the token that pass from client allow the user to change the password

File: src/controllers/auth.ts [hunk 1]
```diff
@@ -69,3 +72,79 @@ export const login = async (req: Request, res: Response): Promise<void> => {
...
+    return
+  }
+
+  const token = tenant.resetPasswordToken()
+  const linkText = 'click here to reset your password'
+  const { name } = tenant
+  const emailbody = `
+    <p>Hello ${name}：</p>
+    <p>A request has been received to change the password for your Form Builder account.</p>
+    <button><a href="/reset-password?token=${token}">${linkText}</a></button>
```
Comment (smarthulala) [inline src/controllers/auth.ts:132]: the url is a relative url, it will not work. use the full url instead

File: src/controllers/auth.ts [hunk 2]
```diff
@@ -69,3 +72,79 @@ export const login = async (req: Request, res: Response): Promise<void> => {
...
+  }
+  if (newPassword.length < 6) {
+    res.status(StatusCodes.BAD_REQUEST).json({ msg: 'Password must be at least 6 characters' })
+    return
+  }
+  if (newPassword === tenant.password) {
+    res.status(StatusCodes.BAD_REQUEST).json({ msg: 'Please provide a different password' })
+    return
+  }
+  tenant.changedPasswords.push({
```
Comment (smarthulala) [inline src/controllers/auth.ts:96]: ternant.password is undefined. I dont think we need the changed password property in the tenant database

File: src/controllers/auth.ts [hunk 3]
```diff
@@ -69,3 +72,79 @@ export const login = async (req: Request, res: Response): Promise<void> => {
     token,
   })
 }
+export const updatePassword = async (req: Request, res: Response): Promise<void> => {
+  const { newPassword, token } = req.body
```
Comment (smarthulala) [inline src/controllers/auth.ts:76]: you should add error handling for any functions

File: src/controllers/auth.ts [hunk 4]
```diff
@@ -69,3 +72,79 @@ export const login = async (req: Request, res: Response): Promise<void> => {
...
+    createdAt: new Date(),
+  })
+  tenant.password = newPassword
+  await tenant.save()
+  res.status(StatusCodes.OK).json({
+    msg: 'Password updated successfully',
+  })
+  const newToken = tenant.createJwt()
+  tenant.password = undefined
+  res.status(StatusCodes.OK).json({
```
Comment (smarthulala) [inline src/controllers/auth.ts:107]: Is it better to combine two json response into one?

**Review by smarthulala — CHANGES_REQUESTED** [review]

**Pull request closed without merging**